crates/nargo/src/cli/gates_cmd.rs

```diff
@@ -21,6 +23,13 @@ pub fn count_gates_with_path<P: AsRef<Path>>(
     program_dir: P,
     show_ssa: bool,
 ) -> Result<(), CliError> {
+    let (_, crate_type) = lib_or_bin(program_dir.as_ref())?;
+    if crate_type != CrateType::Binary {
+        return Err(CliError::Generic(
+            "It's not possible to count the gates in this crate.\nOnly binary crates have a fixed number of gates.".to_string(),
```

Comment:
I don't understand this error message: "Only binary crates have a fixed number of gates."